src/SFA.DAS.Reservations.Web/Controllers/EmployerReservationsController.cs

```diff
@@ -289,14 +290,42 @@ private string GenerateLimitReachedBackLink(ReservationsRouteModel routeModel)
         [HttpPost]
         [ValidateAntiForgeryToken]
         [Route("{id}/select-course", Name = RouteNames.EmployerSelectCourse)]
-        public async Task<IActionResult> PostSelectCourse(ReservationsRouteModel routeModel, string selectedCourseId)
+        public async Task<IActionResult> PostSelectCourse(ReservationsRouteModel routeModel, PostSelectCourseViewModel postViewModel)
         {
+            if (!ModelState.IsValid)
+            {
+                var cachedReservation = await _mediator.Send(new GetCachedReservationQuery { Id = routeModel.Id.Value });
```

Comment:
This part is the same as above in the get, can we move this into a BuildEmployerSelectCourseViewModel method to remove the duplication of code?

Reply:
change pushed